src/hotspot/share/prims/jvmtiTagMap.cpp

```diff
@@ -2636,120 +2779,81 @@ inline bool VM_HeapWalkOperation::collect_simple_roots() {
   return true;
 }
 
-// Walk the stack of a given thread and find all references (locals
-// and JNI calls) and report these as stack references
-inline bool VM_HeapWalkOperation::collect_stack_roots(JavaThread* java_thread,
-                                                      JNILocalRootsClosure* blk)
+// Reports the thread as JVMTI_HEAP_REFERENCE_THREAD,
+// walks the stack of the thread, finds all references (locals
+// and JNI calls) and reports these as stack references
+inline bool VM_HeapWalkOperation::collect_stack_refs(JavaThread* java_thread,
```

Comment:
It makes sense to refactor the body of the `collect_stack_refs` function by adding 2 functions:
 - collect_virtual_thread_stack_refs
 - collect_platform_thread_stack_refs
 It looks like the same register map can be used for both.